ORK1Kit/ORK1Kit/Common/ORK1DocumentSelectionStepViewController.m

```diff
@@ -0,0 +1,376 @@
+    NSDictionary *views = NSDictionaryOfVariableBindings(headerView, sourceStackView, _continueSkipContainer);
+    ORK1EnableAutoLayoutForViews([views allValues]);
+
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-[headerView]-8-[sourceStackView]-(>=8)-[_continueSkipContainer]-36-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:views]];
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[headerView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:views]];
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-[_continueSkipContainer]-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:views]];
+
+    id viewMarginItem = self.view;
+    if (@available(iOS 11.0, *)) {
+        viewMarginItem = self.view.safeAreaLayoutGuide;
+    }
+    
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:sourceStackView attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationGreaterThanOrEqual toItem:viewMarginItem attribute:NSLayoutAttributeLeading multiplier:1 constant:0]];
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:sourceStackView attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationLessThanOrEqual toItem:viewMarginItem attribute:NSLayoutAttributeTrailing multiplier:1 constant:0]];
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:sourceStackView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:viewMarginItem attribute:NSLayoutAttributeCenterX multiplier:1 constant:0]];
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:sourceStackView attribute:NSLayoutAttributeWidth relatedBy:NSLayoutRelationLessThanOrEqual toItem:self.view.readableContentGuide attribute:NSLayoutAttributeWidth multiplier:1 constant:0]];
+    
+    [NSLayoutConstraint activateConstraints:constraints];
+    
+    self.skipButtonItem = self.internalSkipButtonItem;
+}
+
+- (void)setSkipButtonItem:(UIBarButtonItem *)skipButtonItem {
+    [super setSkipButtonItem:skipButtonItem];
+    _continueSkipContainer.skipButtonItem = skipButtonItem;
+    _continueSkipContainer.optional = self.step.isOptional;
+    _continueSkipContainer.skipEnabled = self.step.isOptional;
+}
+
+- (void)notifyDelegateOnResultChange {
+    [super notifyDelegateOnResultChange];
+    self.skipButtonItem = self.internalSkipButtonItem;
+}
+
+#pragma mark - User interaction
+
+- (void)takePhoto {
+    ORK1WeakTypeOf(self) weakSelf = self;
+    switch ([AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo]) {
+        case AVAuthorizationStatusDenied:
+            [self handleError:[NSError errorWithDomain:NSCocoaErrorDomain code:NSFeatureUnsupportedError userInfo:@{NSLocalizedDescriptionKey:ORK1LocalizedString(@"DOCUMENT_SELECTION_ERROR_NO_CAMERA_PERMISSIONS", nil)}] showSettingsButton:YES];
+            break;
```

Comment:
minor - I think these lines can be omitted and drop to the next case?

Reply:
Good idea. Done